src/utils/rocksdb_serialization.hpp

```diff
@@ -30,6 +30,50 @@ namespace memgraph::utils {
 static constexpr const char *outEdgeDirection = "0";
 static constexpr const char *inEdgeDirection = "1";
 
+namespace {
+struct StartEndPositions {
+  size_t start;
+  size_t end;
+
+  size_t Size() const { return end - start; }
+  bool Valid() const { return start != std::string::npos; }
+};
+
+inline StartEndPositions FindPartStartEndPositions(const std::string_view str, const char delim,
+                                                   unsigned int partNumber) {
+  StartEndPositions result{0, 0};
+  for (int i = 0; i < partNumber; ++i) {
+    result.start = result.end;
+    result.end = str.find(delim, result.start);
+    if (result.end == std::string::npos) {
+      if (i < partNumber - 1) {
+        // We didn't find enough parts.
+        result.start = std::string::npos;
+      }
+      return result;
+    }
+    ++result.end;
+  }
+  return result;
+}
+
+inline std::string_view GetViewOfFirstPartOfSplit(const std::string_view src, const char delimiter) {
+  size_t delimPos = src.find(delimiter);
+  return delimPos == std::string::npos ? src : src.substr(0, delimPos);
+}
+
+inline std::string_view GetViewOfSecondPartOfSplit(const std::string_view src, const char delimiter) {
+  StartEndPositions startEndPos = FindPartStartEndPositions(src, delimiter, 2);
+  return (startEndPos.Valid() ? src.substr(startEndPos.start, startEndPos.Size()) : src);
```

Comment:
Maybe make this a helper function or even return this directly from FindPartStartEndPositions.

Reply:
Also, is this the behavior we want; return everything if part wasn't found?

Reply:
We discussed with Andi about this here: https://github.com/memgraph/memgraph/pull/1155#discussion_r1289962009
I think this is what we want now. Because some cases expect this.